nubilanding/src/components/blog/form.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import styles from './styles.module.scss';
+
+class Form extends React.Component {
+  state = { email: '' };
+  constructor(args) {
+    super(args)
+    this.state = {
+
+    }
+  }
+
+  onChange(e) {
+    this.setState({
+      [e.target.email]: e.target.value
+    })
+  }
+  render() {
+    return (
+      <form className={styles.formContenedor}>
+        <input
+          type="email"
+          name="email"
+          placeholder="Dejanos tu email"
+          value={this.state.email}
+          onChange={this.onChange.bind(this)}
```

Comment:
Esto viene de la mano con lo del state duplicado:

Al declarar el state con constructor es necesario hacer los bindings de las funciones ya que las mismas no saben que es "this", pero hay una forma en la que se realizan automáticamente.

Para eso tenés que declarar tus funciones como arrow functions (**=>**) así no necesitas hacer un bind del "this" sino que infiera el this de su contexto de ejecución. Acá tenés una [explicación más detallada](https://medium.com/@nikolalsvk/loosing-bind-this-in-react-8637ebf372cf)

```JSX
class Form extends React.Component {
  state = { email: ' ' }

  onChange = event => this.setState({ email: e.target.value })

  render() {
  ...
  onChange={this.onChange}
```